### PR TITLE
Feature: Add iFlyTek Spark LLM chat model support

### PR DESCRIPTION
**Pull request by vsxd**
- **Description:** This PR enables LangChain to access the iFlyTek's Spark LLM via the chat_models wrapper.
  - **Dependencies:** websocket-client ^1.6.1
  - **Tag maintainer:** @baskaryan 

### SparkLLM chat model usage

Get SparkLLM's app_id, api_key and api_secret from [iFlyTek SparkLLM API Console](https://console.xfyun.cn/services/bm3) (for more info, see [iFlyTek SparkLLM Intro](https://xinghuo.xfyun.cn/sparkapi) ), then  set environment variables `IFLYTEK_SPARK_APP_ID`, `IFLYTEK_SPARK_API_KEY` and `IFLYTEK_SPARK_API_SECRET` or pass parameters when using it like the demo below:

```python3
from langchain.chat_models.sparkllm import ChatSparkLLM

client = ChatSparkLLM(
    spark_app_id="<app_id>",
    spark_api_key="<api_key>",
    spark_api_secret="<api_secret>"
)
```

### CODE REVIEW

**Comment by vsxd** [comment]
lgtm

File: libs/langchain/pyproject.toml
Comment (efriis) [inline libs/langchain/pyproject.toml:1]: Revert to master for this file. This would only be for unit tests, and you only have integration tests.

File: libs/langchain/poetry.lock
Comment (efriis) [inline libs/langchain/poetry.lock:1]: Same - revert to master

**Review by efriis — COMMENTED** [review]
Can you
- add documentation
- revert the dependency change or add unit tests that rely on the new dep

I can help with some of the following because it's new, but many of the imports in `sparkllm.py` should be imported from `langchain_core` instead of `langchain` now. See some other chat models for reference of how to import!

File: libs/langchain/poetry.lock
Comment (vsxd) [reply libs/langchain/poetry.lock:1]: thanks for your guide. I've revert the `pyproject.toml` and `poetry.lcok`

**Comment by vsxd** [comment]
I wrote this code before the refactor, so I missed these import changes. 

I've fixed the import issue, could you please review the PR again? thank you. @efriis 

btw, after the langchain_core refactor, I always get errors in my local unit tests, is this my local env problem?

**Comment by vsxd** [comment]
@efriis could you please review the PR and trigger the workflow? thank you~

**Comment by efriis** [comment]
Hey @vsxd ! This won't be reviewed until next week because of some ongoing refactoring in these folders. Will follow up next week, and may require some small changes to address merge conflicts.

File: libs/langchain/langchain/chat_models/sparkllm.py
Comment (efriis) [inline libs/langchain/langchain/chat_models/sparkllm.py:372]: Why is `ssl.CERT_NONE` required here? Are there certificate issues on the domain?

File: libs/langchain/langchain/chat_models/sparkllm.py
Comment (efriis) [inline libs/langchain/langchain/chat_models/sparkllm.py:478]: Instead of this, raise a more relevant Python error, such as `ConnectionError`

File: libs/langchain/tests/integration_tests/chat_models/test_sparkllm.py
Comment (efriis) [inline libs/langchain/tests/integration_tests/chat_models/test_sparkllm.py:54]: Could you make this a unit test instead?

**Review by efriis — COMMENTED** [review]
Ok the refactor is in @vsxd ! If you want to move it over to `community` like the other chat models, that would be great!

File: libs/langchain/langchain/chat_models/sparkllm.py
Comment (vsxd) [reply libs/langchain/langchain/chat_models/sparkllm.py:372]: I had certificate issue before with their api, but haven't had this problem recently. I'll remove the parameter and test it, then submit my fix.

File: libs/langchain/tests/integration_tests/chat_models/test_sparkllm.py
Comment (vsxd) [reply libs/langchain/tests/integration_tests/chat_models/test_sparkllm.py:54]: If I want to add unit tests, I have to modify `pyproject.toml`. You previously asked me to remove extra dependency to maintain consistency with the master branch, I would like to be able to modify or remove this integration test, is that possible?

Comment (vsxd) [reply libs/langchain/tests/integration_tests/chat_models/test_sparkllm.py:54]: I'm removing this integration test for now, I'll make changes if you feel the existing integration tests is insufficient or if I need to add additional unit tests. Thank you~🙏

**Comment by vsxd** [comment]
> Ok the refactor is in @vsxd ! If you want to move it over to `community` like the other chat models, that would be great!

What I need to do is copy my chat model to `community` or move it? I see that other chat models exist in both modules.

**Comment by efriis** [comment]
If you look at the ones in `langchain`, they're just stubs referencing the community ones! You should do the same, and only export the main classes from `langchain` (some other ones export protected members for backwards compatibility)

**Comment by vsxd** [comment]
> If you look at the ones in `langchain`, they're just stubs referencing the community ones! You should do the same, and only export the main classes from `langchain` (some other ones export protected members for backwards compatibility)

got it, I will do that, thanks.

**Comment by vsxd** [comment]
Hey @efriis , I'm done. Do you have time for another review? Thank you so much!  btw, why did the vercel deplyment fail?

**Comment by vsxd** [comment]
@efriis Are you available for a code review? hope to merge this week🤓

**Comment by vsxd** [comment]
Hey @efriis  , happy new year🎉 Are you available to do another review?

**Comment by vsxd** [comment]
@efriis Hi, Do you have time for a review? Would like to merge this PR if possible.

**Comment by shell-nlp** [comment]
> @efriis Hi, Do you have time for a review? Would like to merge this PR if possible. 您好，请问您有时间审核吗？如果可能的话想合并这个 PR。

太赞了，我也准备支持 spark 呢，发现了你的PR ，又可以当伸手党了

**Comment by vsxd** [comment]
@baskaryan Would you mind reviewing this PR and merge it if possible?

**Review by baskaryan — COMMENTED** [review]
can we remove the changes to `libs/langchain`? the models that are re-exported from there are only included for backwards compatibility (because they used to be in libs/langchain). for new code you only need to update lib/community. we'll be deleted all the re-exports from libs/langchain in the next minor release anyways

**Comment by vsxd** [comment]
> can we remove the changes to `libs/langchain`? the models that are re-exported from there are only included for backwards compatibility (because they used to be in libs/langchain). for new code you only need to update lib/community. we'll be deleted all the re-exports from libs/langchain in the next minor release anyways

@baskaryan Thanks for your help, I'm done ;)

**Review by baskaryan — APPROVED** [review]

**Pull request merged**
